core/ArchiveProcessor/Loader.php

```diff
@@ -119,20 +123,77 @@ private function prepareArchiveImpl($pluginName)
         // invalidate existing archives before we start archiving in case data was tracked in the past. if the archive is
         // made invalid, we will correctly re-archive below.
         if ($this->invalidateBeforeArchiving
-            && Rules::isBrowserTriggerEnabled()
+          && Rules::isBrowserTriggerEnabled()
         ) {
             $this->invalidatedReportsIfNeeded();
         }
+        // load existing data from archive
+        list($visits, $visitsConverted) = $this->loadArchiveData();
+        if ($this->quickReturn) {
+            return [$visits, $visitsConverted];
+        }
+
+        if (SettingsServer::isArchivePhpTriggered()) {
+            $this->logger->info("initiating archiving via core:archive for " . $this->params);
+        }
+
+        //get Lock ID
+        $lockId = $this->makeArchivingLock();
 
+        //ini lock
+        $this->lock = new LoaderLock($lockId);
+
+        //set mysql lock the entire process if another process is running
+        $this->lock->setLock();
```

Comment:
see https://github.com/matomo-org/matomo/pull/18243#issuecomment-960304570

We only want to get the lock if we are in a root archive, and not if we are archiving (since archiver should already avoid archiving the same thing twice) https://github.com/matomo-org/matomo/pull/18243#issuecomment-960304570 . This way we can avoid using locks when they aren't needed and can potentially also avoid deadlock situations.